packages/now-build-utils/src/fs/node-version.ts

```diff
@@ -26,77 +32,83 @@ export function getLatestNodeVersion(): NodeVersion {
 
 export async function getSupportedNodeVersion(
   engineRange?: string,
-  silent?: boolean
+  isAuto?: boolean
 ): Promise<NodeVersion> {
   let selection = getOldestNodeVersion();
 
-  if (!engineRange) {
-    if (!silent) {
-      debug(
-        'Missing `engines` in `package.json`, using default range: ' +
-          selection.range
-      );
-    }
-  } else {
+  if (engineRange) {
     const found = allOptions.some(o => {
       // the array is already in order so return the first
       // match which will be the newest version of node
       selection = o;
       return intersects(o.range, engineRange);
     });
-    const discontinued = isDiscontinued(selection);
-    if (found && !discontinued) {
-      if (!silent) {
-        debug(
-          'Found `engines` in `package.json`, selecting range: ' +
-            selection.range
-        );
-      }
-    } else {
-      throw new Error(
-        'Found `engines` in `package.json` with an unsupported Node.js version range: ' +
-          engineRange +
-          '\nPlease use one of the following supported ranges: ' +
-          JSON.stringify(supportedOptions.map(o => o.range)) +
-          (discontinued
-            ? '\nThis change is the result of a decision made by an upstream infrastructure provider (AWS).' +
-              '\nRead more: https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html'
-            : '')
-      );
+    if (!found) {
+      const intro = isAuto
+        ? 'This project is using an invalid version of Node.js and must be changed.'
+        : 'Found `engines` in `package.json` with an invalid Node.js version range: ' +
+          engineRange;
+      throw new NowBuildError({
+        code: 'NOW_BUILD_UTILS_NODE_VERSION_INVALID',
+        message:
+          intro +
+          '\n' +
+          pleaseUse +
+          JSON.stringify(supportedOptions.map(o => o.range)),
```

Comment:
How about like this?

```suggestion
          supportedOptions.map(o => JSON.stringify(o.range)).join(', '),
```

Reply:
🤔 the only difference is the square brackets but I kinda like the square brackets 🤓